pkg/daemon/writer.go

```diff
@@ -42,35 +42,56 @@ func NewNodeStateStatusWriter(c snclientset.Interface, n string, f func(), devMo
 	}
 }
 
-// Run reads from the writer channel and sets the interface status. It will
-// return if the stop channel is closed. Intended to be run via a goroutine.
-func (writer *NodeStateStatusWriter) Run(stop <-chan struct{}, refresh <-chan Message, syncCh chan<- struct{}, destDir string, runonce bool, platformType utils.PlatformType) {
-	glog.V(0).Infof("Run(): start writer")
+// RunOnce initial the interface status for both baremetal and virtual environments
+func (writer *NodeStateStatusWriter) RunOnce(destDir string, platformType utils.PlatformType) error {
+	glog.V(0).Infof("RunOnce(): start writer")
 	msg := Message{}
 
-	var err error
-
 	if platformType == utils.VirtualOpenStack {
-		writer.metaData, writer.networkData, err = utils.GetOpenstackData()
+		ns, err := writer.getCheckPointNodeState(destDir)
 		if err != nil {
-			glog.Errorf("Run(): failed to get OpenStack data: %v", err)
+			return err
+		}
+
+		metaData, networkData, err := utils.GetOpenstackData()
+		if err != nil {
+			glog.Errorf("RunOnce(): failed to read OpenStack data: %v", err)
 		}
-	}
 
-	if runonce {
-		glog.V(0).Info("Run(): once")
-		if err := writer.pollNicStatus(platformType); err != nil {
-			glog.Errorf("Run(): first poll failed: %v", err)
+		if ns == nil {
+			writer.openStackDevicesInfo, err = utils.CreateOpenstackDevicesInfo(metaData, networkData)
+			if err != nil {
+				return err
+			}
+		} else {
+			devicesInfo := make(utils.OSPDevicesInfo)
```

Comment:
nit: can we move this logic to: `utils.CreateOpenstackDevicesInfoFromNodeStatus(ns)` or similar ?